samples/nrf9160/cloud_client/src/main.c

```diff
@@ -60,12 +78,23 @@ void cloud_event_handler(const struct cloud_backend *const backend,
 		break;
 	case CLOUD_EVT_READY:
 		LOG_INF("CLOUD_EVT_READY");
+
+		k_delayed_work_cancel(&connect_work);
+
 #if defined(CONFIG_CLOUD_PUBLICATION_SEQUENTIAL)
 		k_delayed_work_submit(&cloud_update_work, K_NO_WAIT);
 #endif
 		break;
 	case CLOUD_EVT_DISCONNECTED:
 		LOG_INF("CLOUD_EVT_DISCONNECTED");
+
+		cloud_connected = false;
```

Comment:
nit: This variable is introduced in the following commit, so it will break bisectability. Would be good to move this line to https://github.com/nrfconnect/sdk-nrf/pull/3194/commits/18f6855318e605a4565eaa011ae2e7b442ebb226

Reply:
Done, thanks.